### PR TITLE
Fix builtin and keywords highlighting depending on indentation

### LINKED ISSUE #2159: Built-in 'print' statement is displayed as two different colors depending on indentation
_From [doug.t...@gmail.com](https://code.google.com/u/107157864673511712940/) on 2015-02-05T17:05:42Z_

Summary:
The built-in 'print' statement is displayed as two different colors (in Spyder/Dark theme: blue, purple) depending on indentation. It appears that the issue is because "print" belongs to both keywords.kwlist and dir(__builtins__)

Spyder Version:  2.3.2
Python Version:  2.7.6
Qt Version    :  4.8.4, PyQt4 (API v2) 4.9.6 on Windows
IPython >=0.13  :  1.1.0 (OK)
coverage >=0.25 :  3.7.1 (OK)
matplotlib >=1.0:  1.3.1 (OK)
pandas >=0.13.1 :  0.14.1 (OK)
pep8 >=0.6      :  1.4.6 (OK)
pyflakes >=0.5.0:  0.7.3 (OK)
pygments >=1.6  :  1.6 (OK)
pylint >=0.25   :  1.0.0 (OK)
rope >=0.9.2    :  0.10.2 (OK)
sphinx >=0.6.6  :  1.2 (OK)
sympy >=0.7.0   :  0.7.4.1 (OK)
zmq >=2.1.11    :  14.0.1 (OK)

To Reproduce:
1. In an empty .py file, type 'print' on line 1 and '    print' on line two.

Expected Output: both words are the same color
Actual Output: line 1 is color coded as a "keyword" and line 2 is color coded as a "builtin".

Additional Info:
I checked all the default themes for the editor - each one has the same issue. I also checked all of the python keywords as given by ``keywords.kwlist`` and built-in functions given by ``dir(__builtins__)``. Only "print" has this issue, though other builtins are not colored as such.

I've attached a file which you can load into Spyder and see for yourself.

The purpose of this issue is to choose a color (builtin or keyword) for "print" and use it in all cases. I would prefer it being labeled as a builtin.

**Attachment:** [print_keyword_color_discrepency_example.py](http://code.google.com/p/spyderlib/issues/detail?id=2159)

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=2159_

### PR DESCRIPTION
**Pull request by mariacamilaremolinagutierrez**
Fixes #2159

### CODE REVIEW

**Comment by mariacamilaremolinagutierrez** [comment]
This is what used to happen:
image

Now it is corrected: 
image

File: spyder/utils/syntaxhighlighters.py [hunk 1]
```diff
@@ -281,9 +281,13 @@ def any(name, alternates):
 
 def make_python_patterns(additional_keywords=[], additional_builtins=[]):
     "Strongly inspired from idlelib.ColorDelegator.make_pat"
-    kw = r"\b" + any("keyword", keyword.kwlist+additional_keywords) + r"\b"
+    kwlist = keyword.kwlist+additional_keywords
```
Comment (ccordoba12) [inline spyder/utils/syntaxhighlighters.py]: Please add spaces around the `+` sign here

**Comment by ccordoba12** [comment]
Hey, nice improvement @mariacamilaremolinagutierrez!

Does this also fixes the problem with `print` shown in different colors, as reported on issue #2159?

**Comment by ccordoba12** [comment]
Also, could explain what was the problem before? I mean, why coloring was not applied correctly?

**Comment by ccordoba12** [comment]
And more thing: you forgot to branch 3.x in your fork :-)

I mean, you directly pushed your commits on the 3.x branch of your branch, while you should have created a branch after checking out 3.x :-)

**Comment by mariacamilaremolinagutierrez** [comment]
The problem was that the words True, False and None were in both lists: keywords and builtins. I left them to be colored as builtins. 

The print is not in both lists, so as another member commented on the issue, print was not a problem anymore, at least not in python 3. 

Next time I'll create a new branch. My apologies!

**Comment by ccordoba12** [comment]
Great, thanks a lot @mariacamilaremolinagutierrez!

**Pull request merged**